pandas/tests/computation/test_eval.py

```diff
@@ -600,6 +600,18 @@ def test_float_comparison_bin_op(self, dtype, expr):
         res = df.eval(expr)
         assert res.values == np.array([False])
 
+    def test_unary_in_function(self):
+        # GH 46471
+        df = DataFrame({"x": [0, 1, np.nan]})
+
+        result = df.eval("x.fillna(-1)")
+        expected = df.x.fillna(-1)
+        tm.assert_series_equal(result, expected, check_names=False)
```

Comment:
Why `check_names=False`?

Reply:
Weird, the name check passed locally, but failed on not single cpu and on CircleCI.

See the results of my previous commit [here](https://github.com/pandas-dev/pandas/actions/runs/3034645554/jobs/4883948812).

Reply:
> Weird, the name check passed locally, but failed on not single cpu and on CircleCI.
> 
> See the results of my previous commit [here](https://github.com/pandas-dev/pandas/actions/runs/3034645554/jobs/4883948812).

Seems it's caused by numexpr engine. After installed locally, the column name becomes `None` using `df.eval`.
I noticed `check_names=False` appears several times in `test_eval.py`.

Reply:
Could we make this check more specific then to only check names when the engine is not numexpr?

Reply:
Changed to `check_names=not USE_NUMEXPR`. Do you think it is OK?